### ISSUE TITLE
[jvm-packages] running spark tests crashes the jvm

### ISSUE THREAD

**Issue by rongou** [dmlc/xgboost#3955]
I’m trying to build the jvm packages, but getting a jvm core dump while running tests.

Ubuntu 18.04
cmake: 3.13.1
maven: 3.5.2
jdk: 1.8.0_191
gcc/g++: 7.3.0

These are the steps I took:

```shell
git clone --recurse-submodules https://github.com/dmlc/xgboost.git
cd xgboost/jvm-packages/
mvn package
```
Here is the log output.
[log.txt](https://github.com/dmlc/xgboost/files/2634065/log.txt)

**Comment by rongou** [comment]
BTW this only happens on my desktop with an i7-5820K (6 cores, 12 HT), but not on my laptop with an i7-7600U (2 cores, 4 HT). Wondering if it's some kind of threading issue.

**Comment by CodingCat** [comment]
yeah  it's a known issue in some machines....you can compile with mvn package -DskipTests

**Comment by rongou** [comment]
I see. Where do you think the problem is? In rabit?

**Comment by hcho3** [comment]
Closing for now.

One thing I noticed was that XGBoost4J-Spark tests were susceptible to failure on machines with more than 4 available cores, as by default XGBoost will launch as many worker processes as available cores. That's why the Jenkins CI worker gets access to only 4 cores when running XGBoost4J-Spark tests.
https://github.com/dmlc/xgboost/blob/4e9fad74eb4df6a04758a6b919826388c00530f3/Jenkinsfile#L211-L221

My guess: running multiple processes for Spark in a single local machine may be causing some kind of resource contention.
